### PR TITLE
Configure lowering the first character of fields and cases

### PR DESCRIPTION
**Pull request by friedbrice**
* adds `constructorLowerFirst` to `Options`

* adds `fieldLabelLowerFirst` to `Options`

### CODE REVIEW

File: src/Moat.hs [hunk 1]
```diff
@@ -786,7 +787,7 @@ mkLabel Options {..} =
   AppE (ConE 'Just)
     . stringE
     . fieldLabelModifier
-    . onHeadWith lowerFirstField
+    . bool id (onHeadWith lowerFirstField) fieldLabelLowerFirst
```
Comment (azuzunaga) [inline src/Moat.hs:792]: TIL about the `bool` function. Pretty handy.

**Review by azuzunaga — APPROVED** [review]
Looks good to me, thanks for fixing. There are some warnings about not matching all the cases for `giveBase`, but those seem spurious to me, since the last match, `s -> s` should catch anything that is not a `MoatStruct` or a `MoatEnum`. Maybe because of the as-pattern matching?

**Comment by friedbrice** [comment]
Yeah. They're not spurious, they're legit. Notice each of those lines assumes a singleton list.

But, I'm inclined to leave it, because:
1. it's how I found it, and
2. Moat is used by devs, not customers. If it blows up, we'll find out about it.

**Comment by azuzunaga** [comment]
Ah, I see!

**Comment by tadfisher** [comment]
@friedbrice Is this okay to merge?

**Comment by friedbrice** [comment]
Let me ask @azuzunaga if it's working as intended.

**Comment by friedbrice** [comment]
Good to merge, @tadfisher

File: src/Moat.hs (hunk 1 above)
Comment (friedbrice) [reply src/Moat.hs:792]: TBH, it probably would have been clearer if I'd written an inline `if ... then ... else ...` 😅

**Pull request merged**